connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -252,6 +252,18 @@ class Dataset[T] private[sql] (
     }
   }
 
+  private def containsUnboudRowEncoder(enc: AgnosticEncoder[_]): Boolean = {
```

Comment:
Can we address other nested structures in a follow-up? Granted this covers 90% of the cases. If we end up having a lot of communality with the code in SparkResult, then I'd suggest moving this into a single place.

Reply:
Added [SPARK-44228](https://issues.apache.org/jira/browse/SPARK-44228) to address the nested row encoders all together.